packages/mobile/scripts/run_e2e.sh

```diff
@@ -1,78 +1,105 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
-#####
-# This file launches the emulator and fires the tests
-#####
-
-export CELO_TEST_CONFIG=e2e
-
-adb kill-server && adb start-server
-
-DEFAULT_AVD="Nexus_5X_API_28_x86"
-
-if [[ ! $($ANDROID_SDK_ROOT/emulator/emulator -list-avds | grep ^$DEFAULT_AVD$) ]]; then
-  echo "AVD $DEFAULT_AVD not installed. Pleas install it or change detox' configuration in package.json"
-  echo "You can see the list of available installed devices with $ANDROID_SDK_ROOT/emulator/emulator -list-avds"
-  exit 1
-fi
-
-
-bash ./scripts/unlock.sh
-adb reconnect
-if [ $? -ne 0 ]
-then
-  exit 1
-fi
+# ========================================
+# Build and run the end-to-end tests
+# ========================================
+
+# Flags:
+# -p: Platform (android or ios)
+# -v (Optional): Name of virual machine to run
+# -f (Optional): Fast (skip build step)
+# -r (Optional): Use release build (by default uses debug) 
+# TODO ^ release doesn't work currently b.c. the run_app.sh script assumes we want a debug build
+
+PLATFORM=""
+VD_NAME="Nexus_5X_API_28_x86"
+FAST=false
+RELEASE=false
+while getopts 'p:fr' flag; do
+  case "${flag}" in
+    p) PLATFORM="$OPTARG" ;;
+    v) VD_NAME="$OPTARG" ;;
+    f) FAST=true ;;
+    r) RELEASE=true ;;
+    *) error "Unexpected option ${flag}" ;;
+  esac
+done
 
-echo "Waiting for emulator to unlock..."
-# TODO: improve this to actually poll if the screen is unlocked
-# https://stackoverflow.com/questions/35275828/is-there-a-way-to-check-if-android-device-screen-is-locked-via-adb
-sleep 3
-echo "Emulator unlocked!"
+[ -z "$PLATFORM" ] && echo "Need to set the PLATFORM via the -p flag" && exit 1;
 
-# sometimes the emulator locks itself after boot
-# this prevents that
-bash ./scripts/unlock.sh
+# Ensure jest is accessible to detox
+cp ../../node_modules/.bin/jest node_modules/.bin/
 
 # Just to be safe kill any process that listens on the port 'yarn start' is going to use
 echo "Killing previous metro server (if any)"
-react-native-kill-packager || echo 'Failed to kill for some reason'
-echo "Start metro server"
-yarn react-native start &
-
-
-echo "Waiting for device to connect to Wifi, this is a good proxy the device is ready"
-until adb shell dumpsys wifi | grep "mNetworkInfo" |grep "state: CONNECTED"
-do
-  sleep 10
-done
-
-cp ../../node_modules/.bin/jest node_modules/.bin/
-
-yarn test:detox
-STATUS=$?
-
- # Retry on fail logic
-if [ $STATUS -ne 0 ]; then
-   echo "It failed once, let's try again"
-   yarn test:detox
-   STATUS=$?
-fi
+yarn react-native-kill-packager || echo "Failed to kill package manager, proceeding anyway"
+
+# Build the app and run it
+if [ $PLATFORM = "android" ]; then
+  echo "Using platform android"
+
+  if [ -z $ANDROID_SDK_ROOT ]; then
```

Comment:
I think it should be `$ANDROID_HOME` as recommended in our [eng SETUP](https://github.com/celo-org/celo-monorepo/blob/cd7be93a3abd1333d7972765f23cfa70d17cc492/SETUP.md#install-android-dev-tools) and [React Native setup guide](https://github.com/facebook/react-native-website/blob/067141dca4e4568074bcafb65213aaa95a4eef67/docs/getting-started.md#3-configure-the-android_home-environment-variable)

Reply:
Good catch! Looks like I set both in my bash profile, not sure why